src/blockchain/log-processors/database.ts

```diff
@@ -10,3 +10,25 @@ export function updateMarketState(db: Knex, marketID: Address, trx: Knex.Transac
     db("markets").transacting(trx).update({ marketStateID: marketStateID[0] }).where("marketID", marketID).asCallback(callback);
   });
 }
+
+export function insertPayout(db: Knex, trx: Knex.Transaction, marketID: Address, payoutNumerators: Array<string|number|null>, invalid: boolean, callback: (err: Error|null, payoutID?: number) => void): void {
+  const payoutRow: { [index: string]: string|number|boolean|null } = {
+    marketID,
+    isInvalid: invalid,
+  };
+  payoutNumerators.forEach((value: number, i: number): void => {
+    payoutRow["payout" + i] = value;
+  });
+  db.transacting(trx).select("payoutID").from("payouts").where(payoutRow).first().asCallback( (err: Error|null, payoutID?: number|null): void => {
+    if (err) return callback(err);
+    if (payoutID != null) {
```

Comment:
Is this to stop duplicate insertion by PayoutID?

Since we're writing to the DB Serially this should work, but it is an anti-pattern as its likely to create issues if we ever move to more than one writer at a time (e.g. when we move to Postgres or another DB that supports multiple connections).

It's better to actually create the unique constraint in the DB, do the write and catch the error, so that the synchronization is done at the DB layer, avoiding the race.

Reply:
I agree, i was having difficulty coming up with a pattern that both worked for psql AND sqlite AND returned the insert id. I'll fix it to use replace/on duplicate and select after perhaps.

Reply:
In this specific case, there will not be multiple connections to introduce a race, even in postgres, because of the way we've linearized the processing of logs, but i still agree about fixing it